mass_mailing_partner/models/res_partner.py

```diff
@@ -56,7 +56,8 @@ def write(self, vals):
                 mm_vals['name'] = vals['name']
             if vals.get('email'):
                 mm_vals['name'] = vals['email']
-            self.env["mail.mass_mailing.contact"].search([
+            # Using sudo because ACLs shouldn't produce data inconsistency
+            self.env["mail.mass_mailing.contact"].sudo().search([
```

Comment:
The problem with this is if you track changes, then user who made the change will be incorrect. Why this is needed?

Reply:
Of course, although the change in the `res.partner` object, which is the one the user is actually updating, will be correct.

The explanation is in the description, please read it.